### PR TITLE
Flp 279 tests

### PR DESCRIPTION
**Pull request by ShakilUmar**
image

### CODE REVIEW

File: src/SFA.DAS.Funding.SystemAcceptanceTests.Infrastructure/Events/FinalisedOnProgrammeLearningPaymentEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public static ConcurrentBag<FinalisedOnProgammeLearningPaymentEvent> ReceivedEvents { get; } = new();
+
+    public Task Handle(FinalisedOnProgammeLearningPaymentEvent message, IMessageHandlerContext context)
+    {
+        ReceivedEvents.Add(message);
+        return Task.CompletedTask;
+    }
+
+    public void ClearEvent() => ReceivedEvents.Clear();
```
Comment (tom-gough) [inline src/SFA.DAS.Funding.SystemAcceptanceTests.Infrastructure/Events/FinalisedOnProgrammeLearningPaymentEventHandler.cs:17]: I don't think you are using this method atm, the calling code just calls ReceivedEvents.Clear() directly

File: src/SFA.DAS.Funding.SystemAcceptanceTests/StepDefinitions/CalculateUnfundedPaymentsSetpDefinitions.cs [hunk 2]
```diff
@@ -1,7 +1,8 @@
-﻿using SFA.DAS.Apprenticeships.Types;
-using SFA.DAS.Funding.ApprenticeshipPayments.Types;
+﻿using SFA.DAS.Funding.ApprenticeshipPayments.Types;
 using SFA.DAS.Funding.SystemAcceptanceTests.Helpers;
 using SFA.DAS.Funding.SystemAcceptanceTests.TestSupport;
+using ApprenticeshipCreatedEvent = SFA.DAS.Apprenticeships.Types.ApprenticeshipCreatedEvent;
+using System.Linq;
```
Comment (dinathain) [inline src/SFA.DAS.Funding.SystemAcceptanceTests/StepDefinitions/CalculateUnfundedPaymentsSetpDefinitions.cs:5]: again - think this `using` statement can be cleaned up

File: src/SFA.DAS.Funding.SystemAcceptanceTests.Infrastructure/MessageBus/TestMessageBus.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿using NServiceBus;
+using SFA.DAS.CommitmentsV2.Types;
```
Comment (dinathain) [inline src/SFA.DAS.Funding.SystemAcceptanceTests.Infrastructure/MessageBus/TestMessageBus.cs:2]: and again

File: src/SFA.DAS.Funding.SystemAcceptanceTests/Helpers/WaitHelper.cs [hunk 4]
```diff
@@ -2,7 +2,7 @@
 {
     public static async Task WaitForIt(Func<bool> lookForIt, string failText)
     {
-        var endTime = DateTime.Now.Add(TimeSpan.FromSeconds(20));
+        var endTime = DateTime.Now.Add(TimeSpan.FromSeconds(30));
```
[the remark this answers is not in the document]
Comment (ShakilUmar) [reply src/SFA.DAS.Funding.SystemAcceptanceTests/Helpers/WaitHelper.cs:5]: I think so, With 20 seconds the tests were intermittently to find all the relevant published FinalisedPaymentEvents.

**Review by dinathain — APPROVED** [review]

**Pull request merged**